### PR TITLE
Fix for 'PHP Notice: Undefined offset: 3', which was introduced in sonata-project/GoogleAuthenticator#44

### PR DESCRIPTION
**Pull request by petski**
Fix for 'PHP Notice: Undefined offset: 3', which was introduced in sonata-project/GoogleAuthenticator#44

### CODE REVIEW

File: lib/GoogleAuthenticator.php [hunk 1]
```diff
@@ -103,7 +103,7 @@ public function getUrl($user, $hostname, $secret)
         $args = func_get_args();
         $encoder = 'https://chart.googleapis.com/chart?chs=200x200&chld=M|0&cht=qr&chl=';
         $urlString = '%sotpauth://totp/%s@%s%%3Fsecret%%3D%s%%26issuer%%3D%s';
-        $encoderURL = sprintf($urlString, $encoder, $user, $hostname, $secret, $args[3]);
+        $encoderURL = sprintf($urlString, $encoder, $user, $hostname, $secret, (array_key_exists(3, $args) ? $args[3] : ''));
```
Comment (greg0ire) [inline lib/GoogleAuthenticator.php]: Can you please linebreak this?

Comment (OskarStark) [inline lib/GoogleAuthenticator.php]: and please use the string directly in sprintf instead of the variable, but in a separate commit please!

Comment (petski) [inline lib/GoogleAuthenticator.php]: @greg0ire OK.
@OskarStark I will make the change once it's clear what you mean exactly... Please elaborate.

**Comment by petski** [comment]
Sorry for the incorrect commit message (early monday mornings ...)

@OskarStark Hope new merge now it to your likings
@greg0ire Didn't linebreak because of the changes @OskarStark requested. I think the current style is in line with the rest of the code.

File: tests/GoogleAuthenticatorTest.php [hunk 2]
```diff
@@ -39,10 +39,17 @@ public function testGetCode()
...
         $url = $this->helper->getUrl('foo', 'foobar.org', '3DHTQX4GCRKHGS55CJ', 'FooBar');
         $expected = 'https://chart.googleapis.com/chart?chs=200x200&chld=M|0&cht=qr&chl=otpauth://totp/foo@foobar.org%3Fsecret%3D3DHTQX4GCRKHGS55CJ%26issuer%3DFooBar';
         $this->assertEquals($expected, $url);
     }
+
+    public function testGetUrlNoIssuer()
+    {
+        $url = $this->helper->getUrl('foo', 'foobar.org', '3DHTQX4GCRKHGS55CJ');
+        $expected = 'https://chart.googleapis.com/chart?chs=200x200&chld=M|0&cht=qr&chl=otpauth://totp/foo@foobar.org%3Fsecret%3D3DHTQX4GCRKHGS55CJ';
+        $this->assertEquals($expected, $url);
```
Comment (greg0ire) [inline tests/GoogleAuthenticatorTest.php]: Please avoid one-time variables:
```php
    $this->assertEquals(
        'https://chart.googleapis.com/chart?chs=200x200&chld=M|0&cht=qr&chl=otpauth://totp/foo@foobar.org%3Fsecret%3D3DHTQX4GCRKHGS55CJ',
        $this->helper->getUrl('foo', 'foobar.org', '3DHTQX4GCRKHGS55CJ')
    );
```

Comment (petski) [inline tests/GoogleAuthenticatorTest.php]: I've copied the style from a function that was already there... You are making this the hardest PR I've ever done... :S

**Comment by OskarStark** [comment]
Please fix StyleCI @petski , thank you!

**Comment by greg0ire** [comment]
Please ping us when the Travis build is finished.

**Comment by petski** [comment]
A ping (@greg0ire) as you requested: Travis-CI passed

**Comment by greg0ire** [comment]
By the way, why are you targetting master?

**Comment by greg0ire** [comment]
I think this was a mistake and changed the base to 1.x

**Comment by greg0ire** [comment]
Merged. Thanks @petski !

**Comment by Soullivaneuh** [comment]
Please next time give a changelog.

**Pull request merged**